### PR TITLE
Migrate from ava -> jest 

### PR DESCRIPTION
**Pull request by akleiner2**
## Overview 
More modernizing of the client-side tooling. 

## Testing

- [ ] Updated Typescript types and/or component PropTypes
- [ ] Added / modified component docs
- [ ] Added / modified Storybook stories

### CODE REVIEW

**Review by colinlohner — APPROVED** [review]

File: .eslintrc.json [hunk 1]
```diff
@@ -1,7 +1,8 @@
 {
   "parser": "babel-eslint",
   "env": {
-    "browser": true
+    "browser": true,
+    "jest": true
```
Comment (mshwery) [inline .eslintrc.json:5]: Should we pull this into overrides so it only affects `*.test.*` files?

File: package.json [hunk 2]
```diff
@@ -31,10 +31,11 @@
   ],
   "sideEffects": false,
   "scripts": {
-    "test": "yarn run eslint && ava",
+    "test": "yarn run eslint && yarn run jest",
```
Comment (mshwery) [inline package.json:34]:
```suggestion
    "test": "yarn eslint && yarn jest",
```

File: package.json [hunk 3]
```diff
@@ -31,10 +31,11 @@
...
     "eslint": "eslint 'src/**/*.js'",
     "prepublishOnly": "rm -rf esm commonjs umd && yarn build",
     "dev": "start-storybook -p 6006",
+    "jest": "jest",
```
Comment (mshwery) [inline package.json:38]: This can be removed

**Review by mshwery — APPROVED** [review]
Curious if `npx yarn-deduplicate && yarn install` yields further yarn.lock changes?

File: .eslintrc.json (hunk 1 above)
Comment (akleiner2) [reply .eslintrc.json:5]: Probably - I think a lot of the jest-specific syntax only happens in `*.test.js` files, but we can be double safe 👍

**Pull request merged**